.github/workflows/python-package.yml

```diff
@@ -0,0 +1,35 @@
+name: Python package
+
+on: [push]
+
+env:
+  MIN_CODE_COVERAGE_PERCENT: 93
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install Python
+        uses: actions/setup-python@v4
+        with:
+          python-version-file: '.python-version'
+
+      - name: Install poetry
+        uses: abatilo/actions-poetry@v2
+        with:
+          poetry-version: 1.5.1
+
+      - name: Define a cache for the virtual environment based on the dependencies lock file
+        uses: actions/cache@v3
+        with:
+          path: ./.venv
+          key: venv-${{ hashFiles('poetry.lock') }}
+
+      - name: Install project dependencies
+        run: poetry install --without lint
+
+      - name: Run automated tests (with code coverage)
+        run: poetry run pytest -v --cov=py2puml --cov-branch --cov-report term-missing --cov-fail-under $MIN_CODE_COVERAGE_PERCENT
```

Comment:
Only indirectly related, but a lot better for controlling influence of PRs on overall coverage: https://about.codecov.io/ (free for open source projects and easy integration with gh actions).

Reply:
:pray: I shall give it a look

Reply:
I had a look, thank you for letting me know about codecov. I might integrate it in a later PR but I don't feel the urge for it. The test report with pytest-cov seems good enough for this project now.